Prose/ProseLib/Sources/App/AppReducer.swift

```diff
@@ -30,21 +30,42 @@ public let appReducer: Reducer<
     ),
     Reducer { state, action, environment in
         switch action {
+        case .onAppear:
+            guard case .auth = state.route else { return .none }
+
+            guard let jid = environment.userDefaults.loadCurrentAccount() else { return .none }
+
+            do {
+                guard let password = try environment.credentials.loadCredentials(jid) else { return .none }
+                return Effect(value: .auth(.didLogIn(jid: jid, password: password)))
+            } catch {
+                // TODO: Handle this error, user might have denied access to the credentials
+                print("Error when loading credentials: \(error.localizedDescription)")
+                return .none
+            }
+
         case let .didLogIn(jid):
             state.route = .main(MainScreenState(
                 sidebar: .init(credentials: .init(jid: jid))
             ))
 
-        case let .auth(.didLogIn(jid, _)):
-            let url = URL(staticString: "prose://main")
-            return environment.openURL(url, .init())
-                .receive(on: environment.mainQueue)
-                .map { AppAction.didLogIn(jid: jid) }
-                .mapError(EquatableError.init)
-                .catch { error -> AnyPublisher<AppAction, Never> in
-                    fatalError("Failed to open URL <\(url.absoluteString)>: \(error.localizedDescription)")
-                }
-                .eraseToEffect()
+        case let .auth(.didLogIn(jid, password)):
+            do {
+                environment.userDefaults.saveCurrentAccount(jid)
+                try environment.credentials.save(jid, password)
+
+                let url = URL(staticString: "prose://main")
+                return environment.openURL(url, .init())
+                    .receive(on: environment.mainQueue)
+                    .map { AppAction.didLogIn(jid: jid) }
+                    .mapError(EquatableError.init)
+                    .catch { error -> AnyPublisher<AppAction, Never> in
+                        fatalError("Failed to open URL <\(url.absoluteString)>: \(error.localizedDescription)")
+                    }
+                    .eraseToEffect()
+            } catch {
+                fatalError("Failed to store credentials for '\(jid)': \(error.localizedDescription)")
```

Comment:
No need to crash here. The authentication was successful. A pity that our user will have to enter their credentials again after a restart but why not let them use the app for now?

Reply:
Erm, I'm not sure the app will be 100% usage, won't we need the credentials at other times?

(But you're right, for the moment I'll remove this `fatalError`)

Reply:
Can you elaborate?

Reply:
I mean, will we need to retreive the credentials at other moments than the app startup, to communicate them to ProseCore in order to start new connections for example? Otherwise, how are those connections authenticated? Are the credentials stored in memory in ProseCore (which wouldn't be surprising)?

Reply:
Why would there be more than one connection per account/server?